src/constants/find.rs

```diff
@@ -0,0 +1,520 @@
+    }
+}
+
+impl Into<Find> for Exit {
+    fn into(self) -> Find {
+        match self {
+            Exit::Top => Find::ExitTop,
+            Exit::Right => Find::ExitRight,
+            Exit::Bottom => Find::ExitBottom,
+            Exit::Left => Find::ExitLeft,
+            Exit::All => Find::Exit
+        }
+    }
+}
+
+impl FindConstant for Exit {
+    //TODO: wiarchbe: Check this is correct?
+    type Item = RoomPosition;
+
+    #[inline]
+    fn find_code(&self) -> Find {
+        (*self).into()
+    }
+}
+
+//TODO: wiarchbe: Add back in calculated doc.
+macro_rules! typesafe_find_constants {
+    (
+        $(
+            $vis:vis struct $constant_name:ident = ($value:expr, $result:path);
+        )*
+    ) => (
+        $(
+            #[allow(bad_style)]
+            $vis struct $constant_name;            
+            impl FindConstant for $constant_name {
+                type Item = $result;               
+
+                #[inline]
+                fn find_code(&self) -> Find {
+                    $value
+                }
+            }
+        )*
+    );
+}
+
+typesafe_find_constants! {
+    pub struct CREEPS = (Find::Creeps, Creep);
+    pub struct MY_CREEPS = (Find::MyCreeps, Creep);
+    pub struct HOSTILE_CREEPS = (Find::HostileCreeps, Creep);
+    pub struct SOURCES_ACTIVE = (Find::SourcesActive, Source);
+    pub struct SOURCES = (Find::Sources, Source);
+    pub struct DROPPED_RESOURCES = (Find::DroppedResources, Resource);
+    pub struct STRUCTURES = (Find::Structures, StructureObject);
+    pub struct MY_STRUCTURES = (Find::MyStructures, StructureObject);
+    pub struct HOSTILE_STRUCTURES = (Find::HostileStructures, StructureObject);
+    pub struct FLAGS = (Find::Flags, Flag);
+    pub struct CONSTRUCTION_SITES = (Find::ConstructionSites, ConstructionSite);
+    pub struct MY_SPAWNS = (Find::MySpawns, StructureSpawn);
+    pub struct HOSTILE_SPAWNS = (Find::HostileSpawns, StructureSpawn);
+    pub struct MY_CONSTRUCTION_SITES = (Find::MyConstructionSites, ConstructionSite);
+    pub struct HOSTILE_CONSTRUCTION_SITES = (Find::HostileConstructionSites, ConstructionSite);
+    pub struct MINERALS = (Find::Minerals, Mineral);
+    pub struct NUKES = (Find::Nukes, Nuke);
+    pub struct TOMBSTONES = (Find::Tombstones, Tombstone);
+    pub struct POWER_CREEPS = (Find::PowerCreeps, PowerCreep);
+    pub struct MY_POWER_CREEPS = (Find::MyPowerCreeps, PowerCreep);
+    pub struct HOSTILE_POWER_CREEPS = (Find::HostilePowerCreeps, PowerCreep);
+    pub struct DEPOSITS = (Find::Deposits, Deposit);
+    pub struct RUINS = (Find::Ruins, Ruin);
+    pub struct EXIT_TOP = (Find::ExitTop, RoomPosition);
+    pub struct EXIT_RIGHT = (Find::ExitRight, RoomPosition);
+    pub struct EXIT_BOTTOM = (Find::ExitBottom, RoomPosition);
+    pub struct EXIT_LEFT = (Find::ExitLeft, RoomPosition);
+    pub struct EXIT = (Find::Exit, RoomPosition);
+}
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen]
+    pub type JsFindOptions;
```

Comment:
`JsFindOptions` and `FindOptions` might fit better to implement in room.rs since they're specific to the functions on that object type (their old implementations are still in there commented out at the moment).